### PR TITLE
language filter form & form array fix

### PR DESCRIPTION
**Pull request by pldespaigne**
- [x] create a reusable languages filter component

### CODE REVIEW

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <input test-id="languages-panel-input" matInput type="text" [matAutocomplete]="language"
+      [formControl]="languageControl" placeholder="Add new language"/>
+    <mat-error *ngIf="languageControl.hasError('languageNotSupported')">
+      Please enter a valid language or choose from the dropdown menu!
+    </mat-error>
+    <button matSuffix mat-icon-button (click)="addLanguage(languageControl.value)"
+      [disabled]="!languageControl.valid">
+      <mat-icon svgIcon="magnifying_glasses"></mat-icon>
+    </button>
+    <mat-autocomplete #language="matAutocomplete">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html:15]: I thought we had a auto-complete for static values

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  </button>
+</div>
+
+<!-- LANGUAGE LIST -->
+<div fxLayout="column" fxLayoutAlign="start start">
+  <ng-container *ngIf="selectedLanguages.value">
+    <mat-list [formGroup]="selectedLanguages" role="list">
+      <ng-container *ngFor="let language of (selectedLanguages.valueChanges | async); let i = index">
+        <mat-list-item>
+          <span mat-line>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html:38]: It doesn't work it you put the mat-line on the mat chip directly ?

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  /** Add a new language *(only it's slug)* to the chips list */
+  public addLanguage(language: LanguagesLabel) {
+
+    const languageSlug: LanguagesSlug | null = getCodeIfExists('LANGUAGES', language);
+    if (LANGUAGES_LABEL.includes(language)) {
+      this.selectedLanguages.push(new FormControl(languageSlug));
+      this.analytics.event('addedLanguage', { language });
+    } else {
+      this.languageControl.setErrors({languageNotSupported: true});
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts:51]: this should be managed by the StaticValueForm validator

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      distinctUntilChanged(),
+      debounceTime(300),
+      map(value => this._languageAutocomplete(value)),
+    );
+  }
+
+  /** Add a new language *(only it's slug)* to the chips list */
+  public addLanguage(language: LanguagesLabel) {
+
+    const languageSlug: LanguagesSlug | null = getCodeIfExists('LANGUAGES', language);
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts:46]: You should already have the slug in the option of the autocomplete, why do you need to look for it again ?

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      this.languageControl.setErrors({languageNotSupported: true});
+      // throw new Error('Something went wrong. Please choose a language from the drop down menu.');
+    }
+  }
+
+  public removeLanguage(index: number) {
+    this.selectedLanguages.removeAt(index);
+  }
+
+  public languageFilterChange(event: MatCheckboxChange, language: string, type: 'original' | 'dubbed' | 'subtitle' | 'caption') {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/filters/languages/language-filter.component.ts:60]: Add a comment that reference the todo, this function shoudn't stay like that, it a workaround.

File: libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html (hunk 2 above)
Comment (pldespaigne) [reply libs/movie/src/lib/movie/form/filters/languages/language-filter.component.html:38]: it stretch the mat-chips like a `width: 100%`

**Pull request merged**